File: test/bot_sources/python3/basic_bot.py
# -*- coding: utf-8 -*-
# Python3.4*

import fileinput
import random

if __name__ == '__main__':
    for line in fileinput.input():
        stripped = line.strip()
        if len(stripped) == 0:
            continue
            
        words = stripped.split(' ')
        if words[0] != 'action':
            continue

        move = random.choice(['rock', 'paper', 'scissors'])
        print(move)

File: test/bot_sources/go/basic_bot.go
package main

import (
	"bufio"
	"fmt"
	"math/rand"
	"os"
	"strings"
)

func main() {
	actions := [3]string{"rock", "paper", "scissors"}
	scanner := bufio.NewScanner(os.Stdin)

	for scanner.Scan() {
		line := strings.TrimSpace(scanner.Text())
		words := strings.Split(line, " ")

		if words[0] != "action" {
			continue
		}

		choice := rand.Intn(3)
		fmt.Println(actions[choice])
	}
}
